skbio/io/format/blast7.py

```diff
@@ -0,0 +1,387 @@
+    #   -First line contains "BLAST"
+    #   -Second line contains "Query" or "Database"
+    #   -Third line starts with "Subject" or "Query" or "Database"
+    lines = [line for _, line in zip(range(3), fh)]
+    if len(lines) < 3:
+        return False, {}
+
+    if not lines[0].startswith("# BLAST"):
+        return False, {}
+    if not (lines[1].startswith("# Query:") or
+            lines[1].startswith("# Database:")):
+        return False, {}
+    if not (lines[2].startswith("# Subject:") or
+            lines[2].startswith("# Query:") or
+            lines[2].startswith("# Database:")):
+        return False, {}
+
+    return True, {}
+
+
+@blast7.reader(pd.DataFrame, monkey_patch=False)
+def _blast7_to_data_frame(fh):
+    line_num = 0
+    columns = None
+    skiprows = []
+    for line in fh:
+        if line == "# Fields: \n":
+            # Identifies Legacy BLAST 9 data
+            line = next(fh)
+            line_num += 1
+            if columns is None:
+                columns = _parse_fields(line, legacy=True)
+                skiprows.append(line_num)
+            else:
+                next_columns = _parse_fields(line, legacy=True)
+                if columns != next_columns:
+                    raise BLAST7FormatError("Fields %r do not equal fields %r"
+                                            % (columns, next_columns))
+                skiprows.append(line_num)
+        elif line.startswith("# Fields: "):
+            # Identifies BLAST+7 data
+            if columns is None:
+                columns = _parse_fields(line)
+            else:
+                # Affirms data types do not differ throught file
+                next_columns = _parse_fields(line)
+                if columns != next_columns:
+                    raise BLAST7FormatError("Fields %r do not equal fields %r"
+                                            % (columns, next_columns))
+        line_num += 1
+    if columns is None:
+        # Affirms file contains BLAST data
+        raise BLAST7FormatError("File contains no BLAST data.")
+    fh.seek(0)
+
+    return _parse_blast_data(fh, columns, BLAST7FormatError,
+                             "Number of fields (%r) does not equal number"
+                             " of data columns (%r).", comment='#',
+                             skiprows=skiprows)
+
+
+def _parse_fields(line, legacy=False):
+    """Removes '\n' from fields line and returns fields as a list (columns)."""
+    line = line.rstrip('\n')
+    if legacy:
+        fields = line.split(',')
+    else:
+        line = line.split('# Fields: ')[1]
+        fields = line.split(', ')
+    columns = []
+    for field in fields:
+        if field not in column_converter:
+            raise BLAST7FormatError("Unrecognized field (%r)."
+                                    " Supported fields: %r"
+                                    % (field,
+                                       set(_possible_columns.keys())))
```

Comment:
This error message should display the supported field names for BLAST+7 format: `set(column_converter.keys())`. Current code is showing the normalized column names.